tools/image_processing/bia-ftplinks/biaftplink.xml

```diff
@@ -1,53 +1,55 @@
 <tool id="bia_download" name="FTP Link for Bioimage Archive" version="@VERSION@+galaxy0" profile="22.05">
-    <description>Download images from Bioimage Archive</description>
+    <description>Download images (TIFF) from Bioimage Archive</description>
     <macros>
         <import>macros.xml</import>
     </macros>
     <expand macro="requirements" />
     <command detect_errors="aggressive">
 <![CDATA[
-        wget -r 'ftp://ftp.ebi.ac.uk/biostudies/$mode/$path'/Files; 
-        #if '$ftp_output'
-            #set study = $path.split('/')[-1].rstrip('/')
-            curl https://www.ebi.ac.uk/biostudies/api/v1/studies/$study/info -s |jq -r .ftpLink >>ftpLink.txt
-        #end if
+    curl -s https://www.ebi.ac.uk/biostudies/api/v1/studies/$accession/info | jq -r .ftpLink >> ftpLink.txt &&
+    wget -q -r -l 0 -i ftpLink.txt &&
+
+    find . -type f -name "*.zip" | while read zip_file; do
+        unzip -o "\$zip_file" -d "\$(dirname "\$zip_file")" &> /dev/null;
+    done
+
 ]]>
     </command>
     <inputs>
-    <param name="mode" type="text" label="Storage mode" help="The storage mode, can be either nfs or fire."/>
-    <param name="path" type="text" label="The path of accession. e.g. S-BIAD/570/S-BIAD570 "/>
+        <param name="accession" type="text" label="The accession ID of BioImages-Core or BioStudies-JCB" help="for eg. S-BIAD570, S-JCBD-201309038"/>
         <param name="ftplink_output" type="boolean" label="Generate FTP links?" help="If set, a file containing FTP links associated with the accession will be generated." />
     </inputs>
     <outputs>
-        <data name="images" format="tiff">
-            <discover_datasets pattern="__name_and_ext__" format="tif,tiff" directory="ftp.ebi.ac.uk" visible="true" recurse="true" />
-        </data>
-        <data format="txt" name="ftplinks" from_work_dir="ftpLink.txt" label="FTP Links">
+        <collection name="images" type="list" label="${tool.name} on ${on_string}: TIFF Images">
+            <discover_datasets pattern="(?P&lt;designation&gt;.*)\.tif" format="tif" directory="ftp.ebi.ac.uk" recurse="true" />
```

Comment:
Do you know if the format is tiff or ome.tiff ?

Reply:
In the UI filtering options, it shows only tif as an option. I kept as it is. 

Btw, there are also several other image formats like lif, lsm, mrc and so on. I skipped them because there is no Galaxy datatype.

Reply:
We can change the UI. The question is if the downloaded files are proper OME.tiff files. https://github.com/galaxyproject/galaxy/blob/5c686c62ecb5c23fa31ebae30db4f5a56c63630a/lib/galaxy/datatypes/images.py#L309

Reply:
how to check that?

Reply:
https://github.com/galaxyproject/galaxy/blob/5c686c62ecb5c23fa31ebae30db4f5a56c63630a/lib/galaxy/datatypes/images.py#L313